File: doc-examples/example-toml-java/src/test/java/example/BookTest.java
package example;

import io.micronaut.serde.ObjectMapper;
import io.micronaut.serde.toml.TomlObjectMapper;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Named;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import static org.junit.jupiter.api.Assertions.assertEquals;

@MicronautTest
public class BookTest {

    @Test
    void testWriteReadBook(@Named(TomlObjectMapper.NAME) ObjectMapper tomlMapper) throws IOException {
        String result = tomlMapper.writeValueAsString(new Book("The Stand", 50));

        Book book = tomlMapper.readValue(result, Book.class);
        assertEquals("The Stand", book.getTitle());
        assertEquals(50, book.getPages());
    }
}

File: doc-examples/example-toml-kotlin/src/test/kotlin/example/BookTest.kt
package example

import io.micronaut.serde.ObjectMapper
import io.micronaut.serde.toml.TomlObjectMapper
import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Named
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

@MicronautTest
class BookTest {

    @Test
    fun testWriteReadBook(@Named(TomlObjectMapper.NAME) tomlMapper: ObjectMapper) {
        val result = tomlMapper.writeValueAsString(Book("The Stand", 50))

        val book = tomlMapper.readValue(result, Book::class.java)!!
        assertEquals("The Stand", book.title)
        assertEquals(50, book.pages)
    }
}
